functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,637 @@
+			err := client.Resource("pods").Namespace(testPodB.GetNamespace()).Delete(context.Background(), testPodB.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Asserting that test-pod-version-1 was labeled with mutation-version: 1")
+		value, ok, err = unstructured.NestedString(testPodB.Object, "metadata", amDef.metadata.kind, amDef.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal("1"))
+	})
+
+	It("should not apply mutations policies after deletion", func() {
+		amDef := &assignMetadataInfo{
+			name:  "mutation-deleted",
+			kinds: []string{"Pod"},
+			scope: "Namespaced",
+			metadata: metadataInfo{
+				kind:  "labels",
+				key:   "mutated",
+				value: "true",
+			},
+		}
+
+		By("Creating the assignMetadata")
+		am := assignMetadataDefinition(amDef)
+		am, err := client.Resource("assignmetadata").Create(context.Background(), am, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating the test-pod-before-delete")
+		testPod := GetTestPod(TestingNamespace)
+		testPod.SetName("test-pod-before-delete")
+		testPod.Unstructured, err = client.Resource("pods").Namespace(testPod.GetNamespace()).Create(context.Background(), testPod.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Asserting that test-pod-before-delete was labeled")
+		value, ok, err := unstructured.NestedString(testPod.Object, "metadata", amDef.metadata.kind, amDef.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeTrue())
+		Expect(value).To(Equal(amDef.metadata.value))
+
+		By("Deleting test-pod-before-delete")
+		err = client.Resource("pods").Namespace(testPod.GetNamespace()).Delete(context.Background(), testPod.GetName(), metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Deleting the assignMetadata")
+		err = client.Resource("assignmetadata").Delete(context.Background(), am.GetName(), metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Creating test-pod-after-delete")
+		testPod = GetTestPod(TestingNamespace)
+		testPod.SetName("test-pod-after-delete")
+		testPod.Unstructured, err = client.Resource("pods").Namespace(testPod.GetNamespace()).Create(context.Background(), testPod.Unstructured, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Asserting that test-pod-after-delete was not labeld by the deleted policy")
+		_, ok, err = unstructured.NestedString(testPod.Object, "metadata", amDef.metadata.kind, amDef.metadata.key)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ok).To(BeFalse())
+
+		By("Deleting test-pod-after-delete")
+		err = client.Resource("pods").Namespace(testPod.GetNamespace()).Delete(context.Background(), testPod.GetName(), metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should be able to match by any match category", func() {
+		By("Creating the test namespaces")
+		includedNamepsace := &unstructured.Unstructured{
```

Comment:
this is a namespace object just use the namespace schema

Reply:
do we want to use 2 clients?